src/lightning/data/streaming/shuffle.py

```diff
@@ -97,55 +97,102 @@ def get_chunks_and_intervals_per_ranks(self, distributed_env: _DistributedEnv, c
         # 2. Shuffle them
         indexes = range(len(chunk_intervals))
 
-        # FIXME: Shuffling should be done only within the nodes to benefit
-        # from cache if the dataset doesn't fit on the node.
-        shuffled_indexes = np.random.RandomState(seed=self.seed + current_epoch).permutation(indexes)
-        shuffled_chunk_intervals = np.asarray(chunk_intervals)[shuffled_indexes]
+        seed_shift = 0 if distributed_env.num_nodes > 1 else current_epoch
```

Comment:
could you add some comments here so that it's clearer why we do these individual things?

Reply:
done